packages/containers/src/ComponentForm/kit/defaultRegistry.js

```diff
@@ -111,10 +111,59 @@ function onError({ errors, error, schema }) {
 	return { errors: getNewErrors(errors, schema, extractErrorMessage(error)) };
 }
 
+/**
+ * Update the content of the properties
+ * @param body is the new value
+ * @param trigger the trigger that call it
+ * @example
+const trigger = {
+	"action":"guessMe",
+	"family":"test",
+	"options":[
+		{
+			"path":"root.updatable_config",
+			"type":"object" // or "string"
+		}
+	],
+	"parameters":[
+		{
+			"key":"arg0.name",
+			"path":"root.updatable_config.name"
+		}
+	],
+	"type":"update"
+};
+ */
+function updateProperties({ body, trigger, properties }) {
+	const newProperties = { ...properties };
+	const targetPath = trigger.options[0].path;
+	const getValue = () => {
+		if (trigger.options[0].type === 'object') {
+			return body;
+		}
+		return body.data;
+	};
+	const targetSplitted = targetPath.split('.');
+	targetSplitted.reduce((acc, current, index) => {
```

Comment:
You don't need to rewrite that : 
```
import { mutateValue } from '@talend/react-forms/lib/UIForm/utils/properties';

const schema = { key: targetPath.split('.') };
const value = trigger.options[0].type === 'object' ? body : body.data;
return {
	properties: mutateValue(properties, schema, value),
};
```

Reply:
please don't mutate external scoped value inside a function that return explicitly said result, it make the code hard to read.